### ISSUE TITLE
New cop idea: `Layout/EmptyLineAfterLeadingRequires`

### ISSUE THREAD

**Issue by r7kamura** [rubocop/rubocop#12400]
## Is your feature request related to a problem? Please describe.

I often see these two patterns of empty lines at the beginning of ruby files, and I feel it would be better if the style of empty lines were unified:

### Pattern 1

```ruby
# frozen_string_literal: true

require 'foo'
bar
baz
```

### Pattern 2

```ruby
# frozen_string_literal: true

require 'foo'

bar
baz
```

## Describe the solution you'd like

How about adding a new cop like `Layout/EmptyLineAfterLeadingRequires` ?

```ruby
# bad
require 'foo'
bar
baz
```

```ruby
# good
require 'foo'

bar
baz
```

The implementation itself is probably easy, so if it looks good, I will create a pull request.

## Describe alternatives you've considered

I once considered whether this cop should also target other `require`s not located at the beginning of the file, but it seems that it would be better not to do so.

**Comment by bquorning** [comment]
Will this cop apply to `require_relative` as well?

**Comment by r7kamura** [comment]
Good point. It certainly seems like it should apply to `require_relative` as well as `require`.

**Comment by fatkodima** [comment]
Should this apply to only leading requires or all requires? I vote for all requires.
And please do open a PR.

**Comment by r7kamura** [comment]
One example of a problem we might have with applying `require`s to other than the top of the file is the following case:

```ruby
a

require 'some_file_for_b'
b

c
```

In the above code, the absence of a blank line indicates that `require` and the following code are related.
However, this is lost if the rule of always having an empty line is applied like this:

```ruby
a

require 'some_file_for_b'

b

c
```

I feel that `require` appearing in the middle of a file is often placed just before a specific related process like this. What do you think about this? Would it be more profitable to regard them as edge cases and create a cop like `Layout/EmptyLineAfterRequires`?

**Comment by sambostock** [comment]
Why not something like this?

```yaml
Layout/EmptyLineAfterRequires:
  Description: 'Checks for an empty line after require statements.'
  Enabled: true
  VersionAdded: '<<next>>'
  EnforcedStyle: top_level_only
  SupportedStyles:
    - all_requires
    - top_level_only
  RequireMethods:
    - require
    - require_relative
    # e.g. rubocop-rails could extend to add require_dependency, for apps using classic autoloader
```
